src/HPolyhedron.jl

```diff
@@ -415,26 +419,42 @@ Concrete linear map of a polyhedron in constraint representation.
 
 ### Output
 
-A polyhedron of the same type as the input (`PT`).
+A polyhedron of the same type as the input if `M` is invertible; otherwise,
```

Comment:
This is not true in general. Imagine `P` is a `Singleton.

Reply:
True, this line should be improved. Since the input types can be quite different, i was thinking in something along these lines

```julia
A set whose type is as "close" as possible to `P`. A polygon (or polytope) in vertex representation if this algorithm has to resort to the vertex representation of `P`; otherwise a set whose type is that same as that of `P`.
```

Reply:
I think you do not understand my point.
If the input is a `Hyperrectangle`, we will create an `HPolytope`. And `Hyperrectangle != HPolytope` :wink:

Here is a suggestion which already incorporates my suggested change for `HPolygon`.

> The result type depends on the input and the approach that was used.
If the input is a `HalfSpace`, then the output is a `HalfSpace`.
If the vertex-based approach was used, then the output is an `VPolytope`.
If the input set is an `AbstractPolygon` , then the output is an `HPolygon`.
If the input set is an `AbstractPolytope` , then the output is an `HPolytope`.
Otherwise, the output is an `HPolyhedron`.